### DISCUSSION TITLE
Rotation of the FreeLoad.RectangularLoad

### DISCUSSION THREAD

**Discussion by DannieCW** [Dlubal-Software/RFEM_Python_Client#272] [Q&A] [answered]
Dear developers, 

I am using the RFem API to create rectangular loads on a surface using the center, side and rotation option:

My load_location input is like this: 
angle = 30
side = 0.4
load_location = [coordinate[0], coordinate[1], side, side, angle]

In the load generated in the RFEM_gui, the location coordinates and sides are presented correctly, but no rotation is taken from the input (see picture below). Would you be able to help me identify how did this happen?
![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/138588349/67b0e2a1-5f0f-46b2-a79a-7637dac3c537)

Thank you. 

Best regards,
Dannie

**Comment by heetrojivadiya** [comment, marked as answer]
Dear Dannie,

We are sorry that you faced the problem. It is due to wrong attribute assignment in code. We will soon correct the code. But meanwhile you can create the FreeLoad angle by giving the additional parameter as dict (ex params={'load_location_rotation' : 30* (pi/180)}) in your FreeLoad.RectangularLoad() for that you have to import pi (```from math import pi```). 

I am sharing my example code. You can compare it and change according into your code.
![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/108512789/a2e95493-657a-438a-8bc1-8d74d07ccee4)

```
from RFEM.enums import *
from math import pi
from RFEM.initModel import Model
from RFEM.BasicObjects.material import Material
from RFEM.BasicObjects.section import Section
from RFEM.BasicObjects.node import Node
from RFEM.BasicObjects.line import Line
from RFEM.BasicObjects.surface import Surface
from RFEM.Loads.freeLoad import FreeLoad
from RFEM.LoadCasesAndCombinations.staticAnalysisSettings import StaticAnalysisSettings
from RFEM.LoadCasesAndCombinations.loadCase import LoadCase

Model(True, "Demo1") # crete new model called Demo1
Model.clientModel.service.begin_modification()

Material(1, 'S235')
Section(1, 'IPE 200')

Node(1, 0.0, 0.0, 0.0)
Node(2, 10, 5.0, 0.0)
Node(3, 10, 10, 0)
Node(4, 0, 5, 0)

Line(1, '1 2')
Line(2, '2 3')
Line(3, '3 4')
Line(4, '4 1')

Surface(1)

NodalSupport(1, '1 4', NodalSupportType.FIXED)

StaticAnalysisSettings.GeometricallyLinear(1, "Linear")
LoadCase.StaticAnalysis(1, 'Self-Weight',analysis_settings_no=1,self_weight=[True, 0.0, 0.0, 1.0])
FreeLoad.RectangularLoad(1, 1,'1', load_magnitude_parameter=[1000], load_location=FreeRectangularLoadLoadLocationRectangle.LOAD_LOCATION_RECTANGLE_CENTER_AND_SIDES, load_location_parameter=[5,5,1,1,0], params={'load_location_rotation' : 30* (pi/180)})

Model.clientModel.service.finish_modification()
```

If you still face any issue then feel free to write.

Thank you for reaching us. We will soon correct the code.

Best regards, 
Heet

**Reply by DannieCW** [reply]
Hi Heet, 

Thanks for the fast response. 
I tried it and now it works perfectly. 

Could you please let me know after the angle is implemented in load_location_parameter? 
Much appreciated. 

Best regards,
Dannie

**Reply by heetrojivadiya** [reply]
Hello @DannieCW,

I am just informing that FreeLoad is updated with new load_location_parameter and you can use it without declaring additional parameters as dict.

Best regards,
Heet

**Reply by DannieCW** [reply]
Hi Heet, 

Thanks for the notice. I made some tests. 
Now I declare the rotation in _load_location_parameter_ with version 1.17.0 of RFEM package.
The rotation angle is not yet presented in this way.
Declaring via _params_ works still. 
Would you be able to find out why?

Best regards,
Dannie

**Reply by heetrojivadiya** [reply]
Hello @DannieCW,

I have checked that v1.17.0 is released earlier than bugfix done in Freeload. You have to wait for new release with updated Freeload to use freeload without declaring additional 'params'. Other way is that you can clone this repo or download RFEM_Python_client repo as .zip and place RFEM folder in your local at your project path.

Best regards,
Heet
